pytorch_lightning/trainer/trainer.py

```diff
@@ -342,9 +345,19 @@ def __init__(
         self.on_gpu = True if (gpus and torch.cuda.is_available()) else False
 
         # tpu config
-        self.on_tpu = num_tpu_cores is not None
-        self.num_tpu_cores = num_tpu_cores
-        assert num_tpu_cores in [1, 8, None], 'num_tpu_cores can only be 1 or 8'
+        if num_tpu_cores is not None:
+            rank_zero_warn("Argument `num_tpu_cores` is now set by `tpu_cores` since v0.7.6"
+                           " and this argument will be removed in v0.9.0", DeprecationWarning)
+
+        if tpu_cores is None:
+            tpu_cores = num_tpu_cores
+        self.on_tpu = tpu_cores is not None
```

Comment:
this is not directly related to this PR, but it is not clear what will happen if a user sets `gpus` and `tpu_cores` in the same time

Reply:
I will check this out and provide an update :]

Reply:
@Borda I ran this on kaggle:
`trainer = pl.Trainer(tpu_cores=[1], gpus=[2], precision=16, max_epochs=20)`
It threw an Exception:
```
MisconfigurationException: 
                You requested GPUs: [2]
                But your machine only has: []
```

Reply:
I know, it was not relaying to this PR comment but more for us as conceptual think how to handle these configurations...
what would be you expected behaviour as a user if you set gpu and tpu?
- take any of them if available
- which has higher priority (to be used) if both are available